.github/workflows/hive-consensus-tests.yml

```diff
@@ -44,19 +46,31 @@ jobs:
     - name: Check if master or release branch
       id: check_conditions
       run: |
-        ref="${{ github.event.workflow_run.head_branch || github.ref }}"
+        ref="${{ github.ref }}"
+        event_name="${{ github.event_name }}"
 
         # Append "refs/heads/" prefix if it's not already there
         if [[ $ref != refs/heads/* ]]; then
           ref="refs/heads/$ref"
         fi
 
+        # Initialize variables
+        skip_docker_build="false"
+        skip_wait_for_docker="false"
+
+        # Set conditions based on branch and event type
         if [[ "$ref" == "refs/heads/master" || $ref == refs/heads/release* ]]; then
```

Comment:
@brbrr 
In case of push we know that other action runs now so we need to wait for it to finish (to not run outdated image few minutes before new one appears).
In case of workflow_dispatch - unless someone hits workflow dispatch in less than 5 minutes after push to master/release branch it should use always most recent image of master/release.

Reply:
right. that's exactly what I said :) 

Ah, I thought we'd skip waiting for the docker action to finish, but that's not true.

Reply:
No, we should wait in case trigger was from release branch commit pushed :) 
In any other scenario we should not wait - we assume docker is already created.